canary/consumer-java/src/main/java/com/amazon/kinesis/video/canary/consumer/CanaryListFragmentWorker.java

```diff
@@ -0,0 +1,114 @@
+        this.region = region;
+        this.credentialsProvider = credentialsProvider;
+        this.streamName = streamName;
+        this.fragmentSelector = fragmentSelector;
+
+        amazonKinesisVideoArchivedMedia = AmazonKinesisVideoArchivedMediaClient
+                .builder()
+                .withCredentials(credentialsProvider)
+                .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(endPoint, region.getName()))
+                .build();
+    }
+
+    public static CanaryListFragmentWorker create(final String streamName,
+                                            final AWSCredentialsProvider credentialsProvider,
+                                            final Regions region,
+                                            final AmazonKinesisVideo amazonKinesisVideo,
+                                            final FragmentSelector fragmentSelector) {
+        final GetDataEndpointRequest request = new GetDataEndpointRequest()
+                .withAPIName(APIName.LIST_FRAGMENTS).withStreamName(streamName);
+        final String endpoint = amazonKinesisVideo.getDataEndpoint(request).getDataEndpoint();
+
+        return new CanaryListFragmentWorker(
+                streamName, credentialsProvider, endpoint, region, fragmentSelector);
+    }
+
+    @Override
+    public List<CanaryFragment> call() {
+        List<CanaryFragment> fragments = new ArrayList<>();
+        try {
+            log.info(MessageFormat.format("Start ListFragment worker on stream {0}", streamName));
+
+            ListFragmentsRequest request = new ListFragmentsRequest()
+                    .withStreamName(streamName).withFragmentSelector(fragmentSelector).withMaxResults(fragmentsPerRequest);
+
+            ListFragmentsResult result = amazonKinesisVideoArchivedMedia.listFragments(request);
+
+
+            log.info(MessageFormat.format("List Fragments called on stream {0} response {1} request ID {2}",
+                    streamName,
+                    result.getSdkHttpMetadata().getHttpStatusCode(),
+                    result.getSdkResponseMetadata().getRequestId()));
+
+            for (Fragment f: result.getFragments()) {
+                fragments.add(new CanaryFragment(f));
+            }
+            String nextToken = result.getNextToken();
+
+            /* If result is truncated, keep making requests until nextToken is empty */
+            while (nextToken != null) {
+                request = new ListFragmentsRequest()
+                        .withStreamName(streamName).withNextToken(nextToken);
+                result = amazonKinesisVideoArchivedMedia.listFragments(request);
+
+                for (Fragment f: result.getFragments()) {
+                    fragments.add(new CanaryFragment(f));
+                }
+                nextToken = result.getNextToken();
+            }
+           
+            fragments.sort(Comparator.comparing(CanaryFragment::getFragmentNumberInt));
```

Comment:
This was a carry over from the KVS example [listFragmentWorker](https://github.com/aws/amazon-kinesis-video-streams-parser-library/blob/master/src/main/java/com/amazonaws/kinesisvideo/parser/examples/ListFragmentWorker.java). Looking at the API docs, I don't see any expected order specified for the return of listFragments(). Sorting would give a consistent order to the list of fragments, making it clear if the fragment numbers are increasing, and by how much each time. This would mostly be useful for debugging; I can remove if not relevant enough for the canary app.

Reply:
Hmm. When we sort, we actually do not know if the fragments were received in monotonic increasing order right? Can we confirm if fragments are expected to be received in monotonically increasing order? If yes, they sorting would mess up with the validation because with every check, we expect the fragment numbers to be increasing.

Reply:
Fragment numbers are expected to increase monotonically based on the ingestion order [[source](https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_reader_Fragment.html#:~:text=This%20value%20monotonically)], but I just found [this](https://docs.aws.amazon.com/cli/latest/reference/kinesis-video-archived-media/list-fragments.html#:~:text=Results%20are%20in%20no%20specific%20order%2C%20even%20across%20pages) confirmation that the returned list from `listFragments()` has no order to it at all. Would you still want to remove the sorting?